var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Variorum(CMakePackage):
+    """Variorum is a library providing vendor-neutral interfaces for
+       monitoring and controlling underlying hardware features.
+    """
+    homepage = "https://variorum.readthedocs.io"
+    git = "https://github.com/llnl/variorum.git"
+    url = "https://github.com/LLNL/variorum/archive/v0.1.0.tar.gz"
+
+    maintainers = ["slabasan", "rountree"]
+
+    version("0.1.0", tag="v0.1.0")
+
+    ############
+    # Variants #
+    ############
+    variant("shared", default=True, description="Build Variorum as shared lib")
+
+    # variants for language support
+    variant("python", default=True, description="Build Variorum Python support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=False, description="Build Variorum's documentation")
+    variant("log", default=False, description="Enable Variorum's logs")
+    variant("debug", default=False, description="Enable Variorum's debug")
+
+    ########################
+    # Package dependencies #
+    ########################
+    depends_on("cmake@2.8:", type="build")
+    depends_on("hwloc@1.11.9")
+
+    #########################
+    # Documentation related #
+    #########################
+    depends_on("py-sphinx", when="+python+docs", type="build")
+
+    root_cmakelists_dir = "../spack-src/src"
```

Comment:
Does this work?
```suggestion
    root_cmakelists_dir = "src"
```

Reply:
The install was failing when `root_cmakelists_dir = src`:
`/tmp/brink2/spack-stage/spack-stage-variorum-0.1.0-7cy7nuiefybcuxg24h65my5p5oteeq2g/spack-build/src`

Spack was looking for the CMakeLists.txt in spack-build/src, but my CMakeLists.txt is in a src directory in the repo.

Reply:
Are you on the latest develop? This is confusing, since all other packages that use `root_cmakelists_dir` simply use the relative path from the root:
```console
$ grep root_cmakelists_dir */package.py
arrow/package.py:    root_cmakelists_dir = 'cpp'
bcl2fastq2/package.py:    root_cmakelists_dir = 'src'
branson/package.py:    root_cmakelists_dir = 'src'
catalyst/package.py:        root_cmakelists_dir."""
catalyst/package.py:        with working_dir(self.root_cmakelists_dir):
catalyst/package.py:        catalyst_source_dir = os.path.abspath(self.root_cmakelists_dir)
catalyst/package.py:    def root_cmakelists_dir(self):
catalyst/package.py:        return join_path(os.path.abspath(self.root_cmakelists_dir),
catalyst/package.py:            os.path.abspath(self.root_cmakelists_dir),
catalyst/package.py:                                   os.path.abspath(self.root_cmakelists_dir)] +
clfft/package.py:    root_cmakelists_dir = 'src'
clhep/package.py:    root_cmakelists_dir = 'CLHEP'  # Extra directory layer.
cpprestsdk/package.py:    root_cmakelists_dir = 'Release'
isaac-server/package.py:    root_cmakelists_dir = 'server'
isaac/package.py:    root_cmakelists_dir = 'lib'
kealib/package.py:    def root_cmakelists_dir(self):
lammps/package.py:    root_cmakelists_dir = 'cmake'
latte/package.py:    root_cmakelists_dir = 'cmake'
llvm/package.py:    def root_cmakelists_dir(self):
mofem-cephas/package.py:    root_cmakelists_dir = 'mofem'
mofem-fracture-module/package.py:    def root_cmakelists_dir(self):
mofem-minimal-surface-equation/package.py:    def root_cmakelists_dir(self):
of-catalyst/package.py:    root_cmakelists_dir = 'src/catalyst'
ompt-openmp/package.py:    root_cmakelists_dir = 'runtime'
pgmath/package.py:    root_cmakelists_dir = 'runtime/libpgmath'
protobuf/package.py:    root_cmakelists_dir = "cmake"
quinoa/package.py:    root_cmakelists_dir = 'src'
revbayes/package.py:    def root_cmakelists_dir(self):
spades/package.py:    root_cmakelists_dir = "src"
talass/package.py:    root_cmakelists_dir = 'StreamingTopology'
tinker/package.py:    root_cmakelists_dir = 'source'
trilinos-catalyst-ioss-adapter/package.py:    root_cmakelists_dir = join_path('packages', 'seacas', 'libraries',
visit/package.py:    root_cmakelists_dir = 'src'
```
Also, the docs explicitly state that the path should be relative to the root, not relative to the build directory: https://spack.readthedocs.io/en/latest/spack.build_systems.html#spack.build_systems.cmake.CMakePackage.root_cmakelists_dir

Reply:
Yep, that was it. I wasn't rebased to the latest develop. `src` works now, thanks!